pkg/gather/clusterconfig/operators_test.go

```diff
@@ -41,5 +59,577 @@ func Test_Operators_Gather(t *testing.T) {
+			}
+			if got2 != tt.want2 {
+				t.Errorf("gatherUnhealthyPods() got2 = %v, want %v", got2, tt.want2)
+			}
+		})
+	}
+}
+
+func Test_Operators_GatherNamespaceEvents(t *testing.T) {
+	type args struct {
+		ctx        context.Context
+		coreClient corev1client.CoreV1Interface
+		namespace  string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []record.Record
+		wantErr bool
+	}{
+		{
+			name: "empty namespace events",
+			args: args{
+				ctx:        context.TODO(),
+				coreClient: kubefake.NewSimpleClientset().CoreV1(),
+				namespace:  "insights-operator",
+			},
+			want:    []record.Record{},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			got, err := gatherNamespaceEvents(tt.args.ctx, tt.args.coreClient, tt.args.namespace)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("gatherNamespaceEvents() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("gatherNamespaceEvents() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_Operators_CollectClusterOperatorResources(t *testing.T) {
+	type args struct {
+		ctx           context.Context
+		dynamicClient dynamic.Interface
+		co            configv1.ClusterOperator
+		resVer        map[string][]string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []clusterOperatorResource
+	}{
+		{
+			name: "empty cluster operator resources",
+			args: args{
+				ctx:           context.TODO(),
+				dynamicClient: dynamicfake.NewSimpleDynamicClient(runtime.NewScheme()),
+				co:            newClusterOperator(),
+				resVer:        map[string][]string{},
+			},
+			want: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := collectClusterOperatorResources(tt.args.ctx, tt.args.dynamicClient, tt.args.co, tt.args.resVer); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("collectClusterOperatorResources() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_Operators_GetOperatorResourcesVersions(t *testing.T) {
+	type args struct {
+		discoveryClient discovery.DiscoveryInterface
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    map[string][]string
+		wantErr bool
+	}{
+		{
+			name:    "empty operator resources versions",
+			args:    args{discoveryClient: kubefake.NewSimpleClientset().Discovery()},
+			want:    map[string][]string{},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			got, err := getOperatorResourcesVersions(tt.args.discoveryClient)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("getOperatorResourcesVersions() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("getOperatorResourcesVersions() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
```

Comment:
I don't consider this type of tests (basically passing empty args and checking no error returned) very useful TBH, but OK... :)

Reply:
Yeh, me neither. I'll address the test improvements in another task, so we can get this one merged to unblock the other one.